client/src/main/java/com/vesoft/nebula/meta/client/async/AsyncMetaClientImpl.java

```diff
@@ -0,0 +1,185 @@
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.facebook.thrift.TException;
+import com.facebook.thrift.async.TAsyncClientManager;
+import com.facebook.thrift.protocol.TBinaryProtocol;
+import com.facebook.thrift.protocol.TProtocolFactory;
+import com.facebook.thrift.transport.TNonblockingSocket;
+import com.facebook.thrift.transport.TNonblockingTransport;
+import com.facebook.thrift.transport.TTransportException;
+import com.google.common.collect.Lists;
+import com.google.common.net.HostAndPort;
+import com.google.common.net.InetAddresses;
+import com.vesoft.nebula.meta.GetPartsAllocReq;
+import com.vesoft.nebula.meta.ListEdgesReq;
+import com.vesoft.nebula.meta.ListSpacesReq;
+import com.vesoft.nebula.meta.ListTagsReq;
+import com.vesoft.nebula.meta.MetaService;
+import com.vesoft.nebula.meta.client.async.entry.GetPartsAllocCallback;
+import com.vesoft.nebula.meta.client.async.entry.ListEdgesCallback;
+import com.vesoft.nebula.meta.client.async.entry.ListSpaceCallback;
+import com.vesoft.nebula.meta.client.async.entry.ListTagsCallback;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Random;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AsyncMetaClientImpl implements AsyncMetaClient {
+    private static final Logger LOGGER =
+        LoggerFactory.getLogger(AsyncMetaClientImpl.class.getName());
+
+    private MetaService.AsyncClient client;
+
+    private TNonblockingTransport transport = null;
+
+    private TAsyncClientManager manager;
+
+
+    private final List<HostAndPort> addresses;
+    private final int connectionRetry;
+    private final int timeout;
+
+    public AsyncMetaClientImpl(List<HostAndPort> addresses, int timeout, int connectionRetry) {
+        checkArgument(timeout > 0);
+        checkArgument(connectionRetry > 0);
+        if (addresses.isEmpty()) {
+            throw new IllegalArgumentException("No meta server address is specified.");
+        }
+
+        addresses.forEach(address -> {
+            String host = address.getHost();
+            int port = address.getPort();
+            if (!InetAddresses.isInetAddress(host) || (port <= 0 || port >= 65535)) {
+                throw new IllegalArgumentException(String.format("%s:%d is not a valid address",
+                    host, port));
+            }
+        });
+
+        this.addresses = addresses;
+        this.connectionRetry = connectionRetry;
+        this.timeout = timeout;
+        if (!connect()) {
+            LOGGER.error("Connection Failed.");
+        }
+    }
+
+    public AsyncMetaClientImpl(String host, int port) {
+        this(Lists.newArrayList(HostAndPort.fromParts(host, port)),
+            DEFAULT_TIMEOUT_MS, DEFAULT_CONNECTION_RETRY_SIZE);
+    }
+
+    public AsyncMetaClientImpl(List<HostAndPort> addresses) {
+        this(addresses, DEFAULT_TIMEOUT_MS, DEFAULT_CONNECTION_RETRY_SIZE);
+    }
+
+    private boolean connect() {
```

Comment:
maybe I think  we should  let user call connect and don't do that in construction